state/factory/workingset.go

```diff
@@ -155,6 +156,10 @@ func (ws *workingSet) runAction(
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, errors.Wrap(action.ErrHitGasLimit, "block gas limit exceeded")
 	}
+	// Reject execution of chainID not equal the node's chainID
+	if elp.ChainID() != config.ChainID() {
```

Comment:
chainID could be loaded from ctx.BlockchainContext

Reply:
+1